Why Elm.md

```diff
@@ -2,32 +2,41 @@
-
-You may wonder how Elm developers can survive with all these restrictions. The strictness of the language opens up other doors, however. Because Elm requires all programs to follow its rules, **the Elm compiler will tell you exactly what's wrong with your program if you make a mistake**. It will tell you if you made an `if/else` expression return two different kinds of values, if you are calling a function with too many arguments, or if you are referencing a variable that does not exist.
+Because Elm requires all programs to follow its rules, **the Elm compiler will tell you exactly what's wrong with your program if you make a mistake**. It will tell you if you made an `if/else` expression return two different kinds of values, if you are calling a function with too many arguments, or if you are referencing a variable that does not exist.
 
 Developing with Elm often feels like an ongoing conversation with the compiler, gently pushing you toward writing better code. Often, you don't even need to look at your application in the browser while implementing a feature — you can trust the compiler to tell you if you are going in the right direction.
 
-You may be asking yourself why you should choose to learn a language that's so different from other tools in the ecosystem. **We'd argue that the experience of working in an alien environment, even if it's just for a little while, can be fruitful.** You are probably not used to dealing with the constraints of purity, immutability, and static typing, which together make up the basic tenets of **functional programming**. These constraints will push you to solve problems in creative ways, and hopefully you will learn strategies that you can employ in other languages.
+
+## Why not Elm?
+
+If you are coming from an object-oriented, dynamically typed language like Ruby or JavaScript, Elm can feel like a straightjacket. There are no `each` or `for` loops in Elm — instead, you use functions like `map`, `filter`, and `fold`. If you need to change a value deep inside a nested data structure, you can't simply read the value and mutate it, as you could in other languages (proponents of the [Law of Demeter](https://en.wikipedia.org/wiki/Law_of_Demeter) might argue that this too is a feature).
+
+Working with widget libraries or mutation-heavy JavaScript tools can be hard due to conflicting paradigms. If you need to use [D3.js](https://d3js.org/) or [Three.js](https://threejs.org/), you won't be happy trying to use Elm at the same time.
+
+While Elm's compiler is one of its main attrations, compiling lots of code takes non-zero time. For applications with hundres of thousands of lines of code, this compilation time (which might reach minutes) might be mildly inconvenient.
```

Comment:
meh, I don't think this paragraph is really necessary. I don't want to scare people off because they think the compiler is slow -- it isn't, unless it's a huge project, and it will be a looooong time before their working at the scale.